In [1]:
import pandas as pd
import numpy as np

In [2]:
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/the-movies-dataset/links_small.csv
/kaggle/input/the-movies-dataset/credits.csv
/kaggle/input/the-movies-dataset/ratings.csv
/kaggle/input/the-movies-dataset/keywords.csv
/kaggle/input/the-movies-dataset/ratings_small.csv
/kaggle/input/the-movies-dataset/movies_metadata.csv
/kaggle/input/the-movies-dataset/links.csv


In [3]:
meta = pd.read_csv('/kaggle/input/the-movies-dataset/movies_metadata.csv')

/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3063: DtypeWarning: Columns (10) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [4]:
meta.head(2)

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   

                               homepage    id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story   862  tt0114709                en   
1                                   NaN  8844  tt0113497                en   

  original_title                                           overview  ...  \
0      Toy Story  Led by Woody, Andy's toys live happily in his ...  ...   
1        Jumanji  When siblings Judy and Peter discover an encha...  ...   

  release_date      revenue runtime  \
0   1995-10-30  373554033.0    81.0   
1   1995-12-15  262797249.0   104.0   

                                    spoken_languages    status  \
0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
1  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   

                                     tagline      title  video vote_average  \
0                                        NaN  Toy Story  False          7.7   
1  Roll the dice and unleash the excitement!    Jumanji  False          6.9   

  vote_count  
0     5415.0  
1     2413.0  

[2 rows x 24 columns]

In [5]:
meta.columns

Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')

In [6]:
meta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

In [7]:
meta[meta['original_title'] != meta['title']][['title', 'original_title']].head()

title                    original_title
28  The City of Lost Children        La Cité des Enfants Perdus
29             Shanghai Triad                         摇啊摇，摇到外婆桥
32           Wings of Courage  Guillaumet, les ailes du courage
57                The Postman                        Il postino
58           The Confessional                  Le confessionnal

In [8]:
(len(meta[meta['revenue'] == 0].index)/len(meta.index))*100 # Percentage of movies where the revenue is 0

83.69330928605991

In [9]:
len(meta[meta['revenue'] != 0].index)

7414

For 7414 movies there is revenue information available. So now what we can do is get a revenue budget ratio, and if revenue is zero the ratio becomes zero.

In [10]:
meta['budget']

0        30000000
1        65000000
2               0
3        16000000
4               0
           ...   
45461           0
45462           0
45463           0
45464           0
45465           0
Name: budget, Length: 45466, dtype: object

In [11]:
meta['budget'] = pd.to_numeric(meta['budget'], errors='coerce')

In [12]:
meta['budget'] = meta['budget'].replace(0, np.nan)
meta['revenue'] = meta['revenue'].replace(0, np.nan)

In [13]:
meta['returns'] = meta['revenue']/meta['budget']

In [14]:
meta.head(2)

adult                              belongs_to_collection      budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000.0   
1  False                                                NaN  65000000.0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   

                               homepage    id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story   862  tt0114709                en   
1                                   NaN  8844  tt0113497                en   

  original_title                                           overview  ...  \
0      Toy Story  Led by Woody, Andy's toys live happily in his ...  ...   
1        Jumanji  When siblings Judy and Peter discover an encha...  ...   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   

     status                                    tagline      title  video  \
0  Released                                        NaN  Toy Story  False   
1  Released  Roll the dice and unleash the excitement!    Jumanji  False   

  vote_average vote_count    returns  
0          7.7     5415.0  12.451801  
1          6.9     2413.0   4.043035  

[2 rows x 25 columns]

In [15]:
len(meta[meta['adult'] == 'True'].index)

9

Since there are only 9 movies that are adult, dropping the information.

In [16]:
meta = meta.set_index("adult")
meta.drop('True', axis = 0)
meta.head(2)

belongs_to_collection      budget  \
adult                                                                  
False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000.0   
False                                                NaN  65000000.0   

                                                  genres  \
adult                                                      
False  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
False  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   

                                   homepage    id    imdb_id  \
adult                                                          
False  http://toystory.disney.com/toy-story   862  tt0114709   
False                                   NaN  8844  tt0113497   

      original_language original_title  \
adult                                    
False                en      Toy Story   
False                en        Jumanji   

                                                overview popularity  ...  \
adult                                                                ...   
False  Led by Woody, Andy's toys live happily in his ...    21.9469  ...   
False  When siblings Judy and Peter discover an encha...    17.0155  ...   

           revenue runtime                                   spoken_languages  \
adult                                                                           
False  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
False  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   

         status                                    tagline      title  video  \
adult                                                                          
False  Released                                        NaN  Toy Story  False   
False  Released  Roll the dice and unleash the excitement!    Jumanji  False   

      vote_average vote_count    returns  
adult                                     
False          7.7     5415.0  12.451801  
False          6.9     2413.0   4.043035  

[2 rows x 24 columns]

In [17]:
meta.reset_index(inplace=True)

In [18]:
meta.head(2)

adult                              belongs_to_collection      budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000.0   
1  False                                                NaN  65000000.0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   

                               homepage    id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story   862  tt0114709                en   
1                                   NaN  8844  tt0113497                en   

  original_title                                           overview  ...  \
0      Toy Story  Led by Woody, Andy's toys live happily in his ...  ...   
1        Jumanji  When siblings Judy and Peter discover an encha...  ...   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   

     status                                    tagline      title  video  \
0  Released                                        NaN  Toy Story  False   
1  Released  Roll the dice and unleash the excitement!    Jumanji  False   

  vote_average vote_count    returns  
0          7.7     5415.0  12.451801  
1          6.9     2413.0   4.043035  

[2 rows x 25 columns]

In [19]:
meta = meta.drop(['adult', 'belongs_to_collection', 'homepage', 'video','imdb_id','poster_path'], axis=1)

In [20]:
meta.head(2)

budget                                             genres    id  \
0  30000000.0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   862   
1  65000000.0  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...  8844   

  original_language original_title  \
0                en      Toy Story   
1                en        Jumanji   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...    21.9469   
1  When siblings Judy and Peter discover an encha...    17.0155   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   

     status                                    tagline      title  \
0  Released                                        NaN  Toy Story   
1  Released  Roll the dice and unleash the excitement!    Jumanji   

   vote_average  vote_count    returns  
0           7.7      5415.0  12.451801  
1           6.9      2413.0   4.043035

In [28]:
import ast
def json_str_to_csv(string, key='name'):
    try:
        d = ast.literal_eval(string)
        return ",".join([each[key] for each in d])
    except Exception as e:
        return ''

In [29]:
json_str_to_csv(meta['genres'][0])

'Animation,Comedy,Family'

In [30]:
meta['genres'] = meta['genres'].apply(json_str_to_csv)
meta['production_companies'] = meta['production_companies'].apply(json_str_to_csv)
meta['spoken_languages'] = meta['spoken_languages'].apply(json_str_to_csv)
meta['production_countries'] = meta['production_countries'].apply(json_str_to_csv)

ValueError: malformed node or string: nan

In [ ]:
meta.head()